app/connector/email/src/main/java/io/syndesis/connector/email/customizer/EMailReceiveCustomizer.java

```diff
@@ -0,0 +1,131 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.connector.email.customizer;
+
+import java.io.IOException;
+import java.util.Map;
+import javax.mail.BodyPart;
+import javax.mail.MessagingException;
+import javax.mail.Multipart;
+import javax.mail.internet.MimeMultipart;
+import org.apache.camel.CamelContext;
+import org.apache.camel.CamelContextAware;
+import org.apache.camel.Exchange;
+import org.apache.camel.Message;
+import org.apache.camel.util.ObjectHelper;
+import org.jsoup.Jsoup;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import io.syndesis.connector.email.EMailConstants;
+import io.syndesis.connector.email.model.EMailMessageModel;
+import io.syndesis.integration.component.proxy.ComponentProxyComponent;
+import io.syndesis.integration.component.proxy.ComponentProxyCustomizer;
+
+public class EMailReceiveCustomizer implements ComponentProxyCustomizer, CamelContextAware, EMailConstants {
+
+    private static final Logger LOG = LoggerFactory.getLogger(EMailReceiveCustomizer.class);
+
+    private CamelContext camelContext;
+
+    @Override
+    public void customize(ComponentProxyComponent component, Map<String, Object> options) {
+        component.setBeforeConsumer(this::beforeConsumer);
+    }
+
+    @Override
+    public void setCamelContext(CamelContext camelContext) {
+        this.camelContext = camelContext;
+    }
+
+    @Override
+    public CamelContext getCamelContext() {
+        return this.camelContext;
+    }
+
+    private void beforeConsumer(Exchange exchange) {
+
+        final Message in = exchange.getIn();
+        final EMailMessageModel mail = new EMailMessageModel();
+
+        if (ObjectHelper.isNotEmpty(in.getBody())) {
+            textFromMessage(in, mail);
+        }
+        if (ObjectHelper.isNotEmpty(in.getHeader(MAIL_SUBJECT))) {
+            mail.setSubject(in.getHeader(MAIL_SUBJECT, String.class));
+        }
+        if (ObjectHelper.isNotEmpty(in.getHeader(MAIL_FROM))) {
+            mail.setFrom(in.getHeader(MAIL_FROM, String.class));
+        }
+        if (ObjectHelper.isNotEmpty(in.getHeader(MAIL_TO))) {
+            mail.setTo(in.getHeader(MAIL_TO, String.class));
+        }
+        if (ObjectHelper.isNotEmpty(in.getHeader(MAIL_CC))) {
+            mail.setCc(in.getHeader(MAIL_CC, String.class));
+        }
+        if (ObjectHelper.isNotEmpty(in.getHeader(MAIL_BCC))) {
+            mail.setBcc(in.getHeader(MAIL_BCC, String.class));
+        }
+        exchange.getIn().setBody(mail);
+    }
+
+    private String getPlainTextFromMultipart(Multipart multipart)  throws MessagingException, IOException{
+        StringBuilder result = new StringBuilder();
```

Comment:
That's true if the email chooses to provide the same text in plaintext & html. However, a lot of email I've looked at puts a "please view this email in a web browser" or "this message requires an html reader" type message in the plaintext part.
The plaintext if-statement includes a break to avoid duplication so will add the same to the html if-statement.